backend/backend-schedule/src/main/java/eu/socialedge/hermes/backend/schedule/domain/Schedule.java

```diff
@@ -54,74 +54,37 @@
     @DBRef @Getter
     private @NotNull Line line;
 
-    /**
-     * @deprecated trips must be separated into outbound and inbound,
-     * since {@link #getInboundTrips()} and {@link #getOutboundTrips()} are
-     * just time wasting operations. Moreover, since one {@link Trip} naturally
-     * can contain stops on any stations (not only from given {@link #line}), the
-     * {@link #matchesRoute(Route, Trip)} will fail to match route for such Trips
-     */
-    @Deprecated
-    private final @NotEmpty List<Trip> trips = new ArrayList<>();
-
-    public Schedule(String id, String description, Availability availability, Line line, List<Trip> trips) {
+    @Getter
+    private final @NotEmpty List<Trip> inboundTrips = new ArrayList<>();
+
+    @Getter
+    private final @NotEmpty List<Trip> outboundTrips = new ArrayList<>();
+
+    public Schedule(String id, String description, Availability availability, Line line, List<Trip> inboundTrips, List<Trip> outboundTrips) {
         this.id = defaultIfBlank(id, UUID.randomUUID().toString());
         this.description = description;
         this.availability = notNull(availability);
         this.line = notNull(line);
-        this.trips.addAll(notEmpty(trips));
+        this.inboundTrips.addAll(notEmpty(inboundTrips));
+        this.outboundTrips.addAll(notEmpty(outboundTrips));
     }
 
-    public Schedule(String description, Availability availability, Line line, List<Trip> trips) {
-        this(null, description, availability, line, trips);
+    public Schedule(String description, Availability availability, Line line, List<Trip> inboundTrips, List<Trip> outboundTrips) {
+        this(null, description, availability, line, inboundTrips, outboundTrips);
     }
 
-    public Schedule(Availability availability, Line line, List<Trip> trips) {
-        this(null, null, availability, line, trips);
+    public Schedule(Availability availability, Line line, List<Trip> inboundTrips, List<Trip> outboundTrips) {
+        this(null, null, availability, line, inboundTrips, outboundTrips);
     }
 
     private Schedule(Builder builder) {
-        this(builder.id, builder.description, builder.availability, builder.line, builder.trips);
+        this(builder.id, builder.description, builder.availability, builder.line, builder.inboundTrips, builder.outboundTrips);
     }
 
     public void setAvailability(Availability availability) {
         this.availability = notNull(availability);
     }
 
-    public boolean addTrip(Trip trip) {
-        return trips.add(trip);
-    }
-
-    public void removeTrip(Trip trip) {
-        trips.remove(trip);
-    }
-
-    public List<Trip> getTrips() {
-        return Collections.unmodifiableList(trips);
-    }
-
-    public List<Trip> getInboundTrips() {
-        return trips.stream()
-            .filter(trip -> matchesRoute(line.getInboundRoute(), trip))
-            .collect(Collectors.toList());
-    }
-
-    public List<Trip> getOutboundTrips() {
```

Comment:
Please add getter, that will return in/out trips as wrapped unmod collections. Schedules are also gonna be modifiable so add/remove methods are also required.